src/backends/torch/native/templates/nbeats.h

```diff
@@ -87,9 +92,25 @@ namespace dd
       {
       }
 
-      SeasonalityBlockImpl(SeasonalityBlockImpl &b)
-          : BlockImpl(b), _bS(b._bS), _fS(b._fS)
+      SeasonalityBlockImpl(const SeasonalityBlockImpl &b)
+          : torch::nn::Module(b), BlockImpl(b), _bS(b._bS), _fS(b._fS)
+      {
+      }
+
+      SeasonalityBlockImpl &operator=(SeasonalityBlockImpl &&b)
```

Comment:
Just curious, would there be a way to have the cloning behavior + operator at higher level once and for all, heritable ? This would allow writing native modules without considering details of having one ore more GPUs.

Reply:
This is a special case where the default operator does not have the right behavior, as explained in the comment below.
Most of the time default operators are sufficient and torch::nn::Cloneable::clone does the job correctly.

The `reset` method is required by `torch::nn::Cloneable` (it is pure virtual) so a native model is forced to be clonable (and thus parallelizable)